src/core/sock/sockinfo_tcp.cpp

```diff
@@ -1003,30 +1086,23 @@ ssize_t sockinfo_tcp::tcp_tx(xlio_tx_call_attr_t &tx_arg)
 
     si_tcp_logfunc("tx: iov=%p niovs=%zu", p_iov, sz_iov);
 
-    size_t total_iov_len =
-        std::accumulate(&p_iov[0], &p_iov[sz_iov], 0U,
-                        [](size_t sum, const iovec &curr) { return sum + curr.iov_len; });
     lock_tcp_con();
 
-    if (cannot_do_requested_dummy_send(m_pcb, tx_arg) ||
-        cannot_do_requested_partial_write(m_pcb, tx_arg, m_b_blocking, total_iov_len) ||
-        tcp_wnd_unavalable(m_pcb, total_iov_len)) {
-        unlock_tcp_con();
-        errno = EAGAIN;
-        return -1;
+    if (cannot_do_requested_dummy_send(m_pcb, tx_arg) || TCP_WND_UNAVALABLE(m_pcb, total_iov_len)) {
```

Comment:
total_iov_len is undefined in case TCP_WND_UNAVALABLE defined.
Do we really need TCP_WND_UNAVALABLE? Do we ever use it?

Reply:
Let's put the total_iov_len in the same place as the DEFINED_TCP_TX_WND_AVAILABILITY exists.

Reply:
Removing this expression in the slow_path might be better since we can allow for partial sends here regardless of the TCP window.